app/services/create_edition_link_monitor.rb

```diff
@@ -0,0 +1,48 @@
+class CreateEditionLinkMonitor
+  def initialize(edition)
+    @edition = edition
+    @uris = extract_links
+  end
+
+  def perform!
+    if can_perform?
+      ActiveRecord::Base.transaction do
+        upsert_resource_monitor
+      end
```

Comment:
Why is the API call transactionalised?

Reply:
Just to protect from mid-call updates, this could be overly defensive though.